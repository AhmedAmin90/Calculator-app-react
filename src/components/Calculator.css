.Calculator {
    width: 75%;
    margin: 10px auto;
    border-radius: 10px;
    padding: 10px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)
}