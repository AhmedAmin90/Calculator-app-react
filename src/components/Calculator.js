import ButtonPanel from './ButtonPanel';
import Display from './Display';
import React, { useState} from 'react';
import PropTypes from 'prop-types';
import calculate from '../logic/calculate';
import './Calculator.css'


const Calculator = ()=> {
    const [value, setValue] = useState({
        total: null,
        next: null,
        operation: null,
      });
      const handleClick = (name) => {
        const data = { ...value };
        const result = calculate(data, name);
        setValue({ ...result });
      };
    
      const { total, next } = value;
      const result = next ? next && next.toString() : total && total.toString();
    return(
        <div className="Calculator">
        <h1>Let's Do Some Math !</h1>
        <Display result={result} />
        <ButtonPanel clickHandler={handleClick} />
        </div>

    )
}

Display.propTypes = {
    result: PropTypes.string,
  };
  
  ButtonPanel.propTypes = {
    clickHandler: PropTypes.func.isRequired,
  };

export default Calculator