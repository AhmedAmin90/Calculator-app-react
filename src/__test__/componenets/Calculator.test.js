/* eslint-disable no-unused-expressions */
import React from 'react';
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Calculator from '../../components/Calculator';

test('Contains heading with text', () => {
  render(<Calculator />);
  const elem = create(<Calculator />);
  expect(elem).toMatchSnapshot();
  const heading = screen.getByText('Lets Do Some Math !');
  expect(heading).toBeInTheDocument;
});
