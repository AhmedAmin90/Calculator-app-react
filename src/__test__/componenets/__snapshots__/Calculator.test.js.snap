exports[`test Contains heading with text 1`] = `
<div
  className="Calculator">
  <h1>
    Lets Do Some Math !
  </h1>
  <div
    className="Display">
    <h1 />
  </div>
  <div>
    <div
      className="Button-panel-row">
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          AC
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          +/-
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          %
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "orange",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          รท
        </button>
      </div>
    </div>
    <div
      className="Button-panel-row">
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          7
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          8
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          9
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "orange",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          X
        </button>
      </div>
    </div>
    <div
      className="Button-panel-row">
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          4
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          5
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          6
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "orange",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          -
        </button>
      </div>
    </div>
    <div
      className="Button-panel-row">
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          1
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          2
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          3
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "orange",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          +
        </button>
      </div>
    </div>
    <div
      className="Button-panel-row">
      <div
        className="Button width-double"
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          0
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "#E0E0E0",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          .
        </button>
      </div>
      <div
        className="Button "
        role="main"
        style={
          Object {
            "backgroundColor": "orange",
          }
        }>
        <button
          className="Button-btn"
          onClick={[Function]}
          type="button">
           
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
